pkg/operator/etcd/pod.go

```diff
@@ -108,6 +112,14 @@ func UpdatePod(obj *corev1.Pod, spec *Spec) {
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&obj.Labels, spec.Labels)
 
+	// Update annotations and recreate pod if annotations are updated or removed.
+	annotations := spec.Annotations
+	annotationsHash := map[string]string{
+		"annotationsDeltaHash": annotationHash(annotations),
+	}
+	update.Annotations(&annotations, annotationsHash)
```

Comment:
This is going to end up mutating the input `spec.Annotations` since maps act like pointers. Let's do something like this:

```go
// Update desired annotations.
update.Annotations(&obj.Annotations, spec.Annotations)
// Record a hash of desired annotation keys to force the Pod
// to be recreated if a key disappears from the desired list.
update.Annotations(&obj.Annotations, map[string]string{
  "planetscale.com/annotation-keys-hash": stringmaps.HashKeys(spec.Annotations),
})
```

Reply:
Done.